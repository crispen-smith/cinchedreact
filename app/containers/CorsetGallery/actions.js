/*
 *
 * CorsetGallery actions
 *
 */

import CORSET_ACTION_TYPES from './constants';

export function defaultAction() {
  return {
    type: CORSET_ACTION_TYPES.DEFAULT_ACTION,
  };
}

export function resetAction() {
  return {
    type: CORSET_ACTION_TYPES.RESET,
  };
}

export function setFilter(filter) {
  return {
    type: CORSET_ACTION_TYPES.SET_FILTER,
    filter,
  };
}

export function installCorsets(corsets) {
  return {
    type: CORSET_ACTION_TYPES.INSTALL_CORSETS,
    corsets,
  };
}
