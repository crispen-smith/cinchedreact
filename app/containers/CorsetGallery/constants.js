/*
 *
 * CorsetGallery constants
 *
 */

export const DEFAULT_ACTION = 'app/CorsetGallery/DEFAULT_ACTION';
