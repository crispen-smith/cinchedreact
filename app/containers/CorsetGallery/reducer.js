/*
 *
 * CorsetGallery reducer
 *
 */

import { fromJS } from 'immutable';
import CORSET_ACTION_TYPES from './constants';

export const initialState = fromJS({});

function corsetGalleryReducer(state = initialState, action) {
  switch (action.type) {
    case CORSET_ACTION_TYPES.SET_FILTER:
      return state.set('filter', action.filter);
    case CORSET_ACTION_TYPES.GET_CORSETS:
      return state;
    case CORSET_ACTION_TYPES.INSTALL_CORSETS:
      return state.setIn(['corsets'], action.corsets);
    default:
      return state;
  }
}

export default corsetGalleryReducer;
