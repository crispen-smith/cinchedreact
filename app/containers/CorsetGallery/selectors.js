import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the corsetGallery state domain
 */

const selectCorsetGalleryDomain = state =>
  state.get('corsetGallery', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by CorsetGallery
 */

const makeSelectCorsetGallery = () =>
  createSelector(selectCorsetGalleryDomain, substate => substate.toJS());

const makeSelectCorsets = () =>
  createSelector(
    selectCorsetGalleryDomain,
    substate =>
      typeof substate.toJS().corsets === 'object'
        ? substate.toJS().corsets
        : [],
  );

const makeSelectCorsetGalleryFilter = () =>
  createSelector(selectCorsetGalleryDomain, substate => substate.toJS().filter);

const makeSelectorFilteredCorsets = () => {
  const selectCorsets = makeSelectCorsets();
  const selectFilter = makeSelectCorsetGalleryFilter();

  return createSelector(
    [selectCorsets, selectFilter],
    (corsets, filter) =>
      /* eslint-disable */
    corsets.filter ?
      filter === 'all'
        ? corsets
        : corsets.filter(corset => corset.type === filter)
      : [],
      /* eslint-enable */
  );
};

export default makeSelectCorsetGallery;

export {
  makeSelectCorsetGallery,
  selectCorsetGalleryDomain,
  makeSelectCorsetGalleryFilter,
  makeSelectorFilteredCorsets,
};
