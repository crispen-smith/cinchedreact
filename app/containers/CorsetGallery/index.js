/**
 *
 * CorsetGallery
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import {
  makeSelectCorsetGallery,
  makeSelectCorsetGalleryFilter,
  makeSelectorFilteredCorsets,
} from './selectors';
import { makeSelectLoggedIn } from '../BrandedHeader/selectors';
import reducer from './reducer';
import saga from './saga';
import * as actions from './actions';

import CorsetThumbnailComponent from '../../components/CorsetThumbnailComponent';
import { ProductGrid, ProductH1 } from '../../resources/ProductStyling';

/* eslint-disable react/prefer-stateless-function */
export class CorsetGallery extends React.Component {
  constructor(props) {
    super(props);
    const { filter } = props.match.params;

    const setFilterAction = actions.setFilter(filter);
    props.dispatch(setFilterAction);
  }
  componentDidMount() {
    this.props.dispatch(actions.resetAction());
  }

  render() {
    const { filter } = this.props;

    const displayFilter =
      filter && filter.substring
        ? filter.substring(0, 1).toUpperCase() +
          filter.substring(1, filter.length)
        : '';

    const displayCorsets =
      this.props.displayCorsets.length >= 1 ? (
        this.props.displayCorsets.map(corset => (
          <CorsetThumbnailComponent
            CorsetName={corset.name}
            CorsetAlt="Create New"
            CorsetThumbnailSource="sample.jpg"
            Summary="Create New"
          />
        ))
      ) : (
        <div>&nbsp;</div>
      );

    const CreateThumbnail = this.props.loggedIn ? (
      <CorsetThumbnailComponent
        CorsetName="Create"
        CorsetAlt="Create New"
        CorsetThumbnailSource="sample.jpg"
        Summary="Create New"
      />
    ) : (
      <div>&nbsp;</div>
    );

    return (
      <div>
        <Helmet>
          <title>{`${displayFilter} Corsets at Cinched Tight`}</title>
          <meta
            name="description"
            content={`${displayFilter} Corsets at Cinched Tight`}
          />
        </Helmet>
        <ProductH1>{displayFilter} Corsets</ProductH1>
        <ProductGrid>
          {displayCorsets}
          {CreateThumbnail}
        </ProductGrid>
      </div>
    );
  }
}

CorsetGallery.propTypes = {
  dispatch: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  displayCorsets: PropTypes.array,
  filter: PropTypes.string,
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  corsetGallery: makeSelectCorsetGallery(),
  filter: makeSelectCorsetGalleryFilter(),
  loggedIn: makeSelectLoggedIn(),
  displayCorsets: makeSelectorFilteredCorsets(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'corsetGallery', reducer });
const withSaga = injectSaga({ key: 'corsetGallery', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(CorsetGallery);
