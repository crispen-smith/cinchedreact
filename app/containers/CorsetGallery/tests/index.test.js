import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import CorsetGallery from '../index';
import configureStore from '../../../configureStore';

describe('<CorsetGallery />', () => {
  it('Renders a Corset Snapshot when LoggedIn = True', () => {
    const brand = { loggedIn: true };
    const match = { params: { filter: 'all' } };
    const store = configureStore({}, browserHistory, brand);
    const gallery = mount(
      <Provider store={store}>
        <CorsetGallery match={match} />
      </Provider>,
    );

    expect(gallery).toMatchSnapshot();
    expect(gallery.find('CorsetThumbnailComponent')).toHaveLength(1);
  });

  it('Renders a Corset Snapshot when LoggedIn = false', () => {
    const brand = { loggedIn: false };
    const match = { params: { filter: 'all' } };
    const store = configureStore({}, browserHistory, brand);
    const gallery = mount(
      <Provider store={store}>
        <CorsetGallery match={match} />
      </Provider>,
    );

    expect(gallery).toMatchSnapshot();
  });
});
