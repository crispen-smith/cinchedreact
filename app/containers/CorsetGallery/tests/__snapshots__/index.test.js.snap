// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CorsetGallery /> Renders a Corset Snapshot when LoggedIn = True 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {
        "corsetGallery": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": null,
            "name": "defaultSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(withSaga(Connect(CorsetGallery)))
    match={
      Object {
        "params": Object {
          "filter": "all",
        },
      }
    }
  >
    <withSaga(Connect(CorsetGallery))
      match={
        Object {
          "params": Object {
            "filter": "all",
          },
        }
      }
    >
      <Connect(CorsetGallery)
        match={
          Object {
            "params": Object {
              "filter": "all",
            },
          }
        }
      >
        <CorsetGallery
          corsetGallery={
            Object {
              "filter": "All",
            }
          }
          dispatch={[Function]}
          displayCorsets={Array []}
          filter="All"
          loggedIn={true}
          match={
            Object {
              "params": Object {
                "filter": "all",
              },
            }
          }
        >
          <div>
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
            >
              <SideEffect(NullComponent)
                defer={true}
                encodeSpecialCharacters={true}
                meta={
                  Array [
                    Object {
                      "content": "All Corsets at Cinched Tight",
                      "name": "description",
                    },
                  ]
                }
                title="All Corsets at Cinched Tight"
                titleAttributes={Object {}}
              >
                <NullComponent
                  defer={true}
                  encodeSpecialCharacters={true}
                  meta={
                    Array [
                      Object {
                        "content": "All Corsets at Cinched Tight",
                        "name": "description",
                      },
                    ]
                  }
                  title="All Corsets at Cinched Tight"
                  titleAttributes={Object {}}
                />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <ProductTitle>
              <ProductTitle__PT>
                <h1
                  className="ProductTitle__PT-iIqmTV bAjZGN"
                >
                  All
                   Corsets
                </h1>
              </ProductTitle__PT>
            </ProductTitle>
            <ProductGrid>
              <ProductGrid__PG>
                <div
                  className="ProductGrid__PG-svRIz bIpvyY"
                >
                  <div>
                     
                  </div>
                  <CorsetThumbnailComponent
                    CorsetAlt="Create New"
                    CorsetName="Create"
                    CorsetThumbnailSource="sample.jpg"
                    Summary="Create New"
                  >
                    <CorsetThumbnailComponent__CTC>
                      <div
                        className="CorsetThumbnailComponent__CTC-bqQMiC iMVBHu"
                      >
                        <a
                          href="/corsets/Create"
                        >
                          <h2>
                            Create
                          </h2>
                          <img
                            alt="Create New"
                            src="http://res.cloudinary.com/www-cinchedtight-com/image/upload/v1533177121/sample.jpg"
                          />
                          <div
                            className="summary"
                          >
                            Create New
                          </div>
                        </a>
                      </div>
                    </CorsetThumbnailComponent__CTC>
                  </CorsetThumbnailComponent>
                </div>
              </ProductGrid__PG>
            </ProductGrid>
          </div>
        </CorsetGallery>
      </Connect(CorsetGallery)>
    </withSaga(Connect(CorsetGallery))>
  </withReducer(withSaga(Connect(CorsetGallery)))>
</Provider>
`;

exports[`<CorsetGallery /> Renders a Corset Snapshot when LoggedIn = false 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {
        "corsetGallery": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 2,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": null,
            "name": "defaultSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(withSaga(Connect(CorsetGallery)))
    match={
      Object {
        "params": Object {
          "filter": "all",
        },
      }
    }
  >
    <withSaga(Connect(CorsetGallery))
      match={
        Object {
          "params": Object {
            "filter": "all",
          },
        }
      }
    >
      <Connect(CorsetGallery)
        match={
          Object {
            "params": Object {
              "filter": "all",
            },
          }
        }
      >
        <CorsetGallery
          corsetGallery={
            Object {
              "filter": "All",
            }
          }
          dispatch={[Function]}
          displayCorsets={Array []}
          filter="All"
          loggedIn={true}
          match={
            Object {
              "params": Object {
                "filter": "all",
              },
            }
          }
        >
          <div>
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
            >
              <SideEffect(NullComponent)
                defer={true}
                encodeSpecialCharacters={true}
                meta={
                  Array [
                    Object {
                      "content": "All Corsets at Cinched Tight",
                      "name": "description",
                    },
                  ]
                }
                title="All Corsets at Cinched Tight"
                titleAttributes={Object {}}
              >
                <NullComponent
                  defer={true}
                  encodeSpecialCharacters={true}
                  meta={
                    Array [
                      Object {
                        "content": "All Corsets at Cinched Tight",
                        "name": "description",
                      },
                    ]
                  }
                  title="All Corsets at Cinched Tight"
                  titleAttributes={Object {}}
                />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <ProductTitle>
              <ProductTitle__PT>
                <h1
                  className="ProductTitle__PT-iIqmTV bAjZGN"
                >
                  All
                   Corsets
                </h1>
              </ProductTitle__PT>
            </ProductTitle>
            <ProductGrid>
              <ProductGrid__PG>
                <div
                  className="ProductGrid__PG-svRIz bIpvyY"
                >
                  <div>
                     
                  </div>
                  <CorsetThumbnailComponent
                    CorsetAlt="Create New"
                    CorsetName="Create"
                    CorsetThumbnailSource="sample.jpg"
                    Summary="Create New"
                  >
                    <CorsetThumbnailComponent__CTC>
                      <div
                        className="CorsetThumbnailComponent__CTC-bqQMiC iMVBHu"
                      >
                        <a
                          href="/corsets/Create"
                        >
                          <h2>
                            Create
                          </h2>
                          <img
                            alt="Create New"
                            src="http://res.cloudinary.com/www-cinchedtight-com/image/upload/v1533177121/sample.jpg"
                          />
                          <div
                            className="summary"
                          >
                            Create New
                          </div>
                        </a>
                      </div>
                    </CorsetThumbnailComponent__CTC>
                  </CorsetThumbnailComponent>
                </div>
              </ProductGrid__PG>
            </ProductGrid>
          </div>
        </CorsetGallery>
      </Connect(CorsetGallery)>
    </withSaga(Connect(CorsetGallery))>
  </withReducer(withSaga(Connect(CorsetGallery)))>
</Provider>
`;
